cylc/flow/scripts/lint.py

```diff
@@ -33,9 +35,19 @@
 
 A non-zero return code will be returned if any issues are identified.
 This can be overridden by providing the "--exit-zero" flag.
+"""
 
-Configurations for Cylc lint can also be set in a pyproject.toml file.
+TOMLDOC = """
+Configurations for Cylc lint can also be set in a pyproject.toml file using
+the following parameters:
 
+{}
+
+   [cylc-lint]                     # any of {}
+       ignore = ['S001', 'S002]    # List of rules to ignore
+       exclude = ['etc/foo.cylc']  # List of files to ignore
+       rulesets = ['style', '728'] # Sets default rulesets to check
+       max-line-length = 130       # Max line length for linting
```

Comment:
Is something supposed to be inserted into the `{}`?

This is what I get:

```console
$ cylc lint --help
...
Configurations for Cylc lint can also be set in a pyproject.toml file using
the following parameters:



   [cylc-lint]                     # any of ['cylc-lint', 'cylclint', 'cylc_lint']
       ignore = ['S001', 'S002]    # List of rules to ignore
       exclude = ['etc/foo.cylc']  # List of files to ignore
       rulesets = ['style', '728'] # Sets default rulesets to check
       max-line-length = 130       # Max line length for linting
...
```

Reply:
Yes, that's what's supposed to happen. Are you going to quote me the "there should be one and preferably only one obvious way"?

Reply:
> Yes, that's what's supposed to happen.

Then what is the `{}` supposed to do?

Reply:
Have the possible toml section headers filled in using str.format()

``` 
        COP_DOC + TOMLDOC.format('', str(LINT_SECTIONS)),
```

Reply:
It doesn't look like anything is being inserted into it, bug?

Reply:
Sorry - there's two `{}` in your snippet, and we're talking about different ones. Sorry.

Reply:
Template is used at line 1116 for the CLI doc and we don't want to insert anything.

It's also used at line 1276 for rst format documentation where an rst directive  is inserted.

Reply:
Gotcha.

It's the blank lines in the CLI output which threw me because it looked like something was missing.

Can shift those newlines into the RST bit:

```diff
​diff --git a/cylc/flow/scripts/lint.py b/cylc/flow/scripts/lint.py
index 1e986fea2..3baa98073 100755
--- a/cylc/flow/scripts/lint.py
+++ b/cylc/flow/scripts/lint.py
@@ -17,7 +17,6 @@
"""Cylc configuration linter.

Checks code style, deprecated syntax and other issues.
-
"""
# NOTE: docstring needed for `cylc help all` output
# (if editing check this still comes out as expected)
@@ -38,11 +37,7 @@ This can be overridden by providing the "--exit-zero" flag.
"""

TOMLDOC = """
-Configurations for Cylc lint can also be set in a pyproject.toml file using
-the following parameters:
-
-{}
-
+pyproject.toml configuration:{}
    [cylc-lint]                     # any of {}
        ignore = ['S001', 'S002]    # List of rules to ignore
        exclude = ['etc/foo.cylc']  # List of files to ignore
@@ -1274,5 +1269,5 @@ def main(parser: COP, options: 'Values', target=None) -> None:
# NOTE: use += so that this works with __import__
# (docstring needed for `cylc help all` output)
__doc__ += TOMLDOC.format(
-    '.. code-block:: toml', str(LINT_SECTIONS)) + get_reference_rst(
+    '\n\n.. code-block:: toml\n', str(LINT_SECTIONS)) + get_reference_rst(
     parse_checks(['728', 'style'], reference=True))
```

Reply:
Yes. Done.